src/lib/components/DashNgl.react.js

```diff
@@ -0,0 +1,253 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Stage, Selection} from 'ngl';
+
+/**
+ * Dash ngl component
+ */
+export default class DashNgl extends Component {
+    // constructor might not be needed anylonger:
+    // https://hackernoon.com/the-constructor-is-dead-long-live-the-constructor-c10871bea599
+    constructor(props) {
+        super(props);
+        this.state = {stage: null, structuresList: []};
+        console.log(this.props);
+        console.log(this.state);
+    }
+
+    // called after the component is rendered
+    componentDidMount() {
+        const {id, stageParameters} = this.props;
+        const params = {...stageParameters};
+        console.log(params);
+
+        const stage = new Stage(id, params);
+
+        this.setState({stage});
+        console.log('component did mount');
+    }
+
+    // triggered by any update of the DOM (e.g. new dropdown selection)
+    shouldComponentUpdate(nextProps) {
+        const {stageParameters} = this.props;
+
+        // when the app is starting data is not correctly interpreted as an array
```

Comment:
Could you elaborate on this? Is `this.props.data` `undefined`, or is it something else?

Reply:
That is a weird bug if you console.log (this.props.data) during the start of the app its properties are changing. See in the last line in the picture below
it somehow changed to true (=array). Since I did not found any explanation for that behaviour I "fixed" that error with this if-else statement.

Do you have a clue why this is happening ?
```
    32         const {id, stageParameters} = this.props;
    33         console.log(this.props)
    34         console.log(this.props.data)
    35         console.log(typeof this.props.data)
    36         console.log(Array.isArray(this.props.data))
```
![console log_thisProps](https://user-images.githubusercontent.com/28399610/76095973-843ce200-5fc5-11ea-8a45-237af51fe5c1.png)

Reply:
resolved by adding a default dictionary :)